core/server/models/plugins/access-rules.js

```diff
@@ -21,12 +21,24 @@ module.exports = function (Bookshelf) {
             return !!(this._context && this._context.public);
         },
 
-        isInternalContext: function isInternalContext() {
+        isInternalContext: function isInternalContext(options) {
+            if (options) {
+                return Model.isInternalContext(options);
+            }
+
             return !!(this._context && this._context.internal);
         }
     },
     {
         /**
+         * Static method to figure out if internal context.
+         * @returns {boolean}
+         */
+        isInternalContext: function isInternalContext(options) {
+            return options && options.context && options.context.internal;
```

Comment:
This feels wrong & I can't quite figure out why it is needed. @kirrg001 could you provide a little explanation?

Reply:
My thinking behind this is to reduce the usage of `if options and options.context` in the code base.
Whenever you want to figure out which context you are in you call

`this.isInternalContext`
`this.isPublicContext`

 (or maybe rename to `isInternalAccess`)

The usage of `isInternalContext` is [here](https://github.com/TryGhost/Ghost/pull/9053/files#diff-f8d02ad12b13d44b912bbf46cf7193b6R179). 
If you update a resource and you are not the internal context (e.g. public, app, external), we have to ensure that you won't override certain fields. 

I've re-used the access plugin, because this plugin already created the idea of having the context checks in one place. BUT this plugin is barely used at the moment, because we almost never pass the options into the `.forge` method (https://github.com/TryGhost/Ghost/pull/9053/files#diff-f02562ed0c95438ac491a4acef430e37L40). I think all of this is going into the correct direction, but this needs a bit more love. I think for now it's useful to outsource the long conditions (if options and if options.context AND...)

FYI: I have added a TODO here https://github.com/TryGhost/Ghost/pull/9069/files#diff-f02562ed0c95438ac491a4acef430e37R48. 

Let me know what you think. The whole context issue also relates to #9043.

Reply:
I _think_ the long term solution to this is to swap context from being a POJO to a class with it's own methods?

Short term:
The approach in this PR gives me a little concern because we now have 2 different method signatures with `x.isInternalContext()` vs `x.isInternalContext(options)` and that it's tricky to know which you should use where. 

However, I'm not sure if long term we want to be passing options to forge, and so it would be correct to do that in lots more places, or whether that is a bad approach and doing that would make this harder to fix later 🤔 

I don't want to hold up these 2 PRs too much, but whilst we both have context just want to see if we can agree on a good long term approach.

Reply:
We agreed on reverting the access rules plugin change for now. We need a real solution for this. It's less confusing for now to have two methods (one static and one who relies on the passed context object on .forge).